packages/nodes-base/nodes/Redis/Redis.node.ts

```diff
@@ -410,6 +433,116 @@ export class Redis implements INodeType {
 				required: true,
 				description: 'Data to publish',
 			},
+			// ----------------------------------
+			//         push/pop
+			// ----------------------------------
+			{
+				displayName: 'List',
+				name: 'list',
+				type: 'string',
+				displayOptions: {
+					show: {
+						operation: [
+							'push',
+							'pop',
+						],
+					},
+				},
+				default: '',
+				required: true,
+				description: 'Name of the list in Redis',
+			},
+			{
+				displayName: 'Data',
+				name: 'messageData',
+				type: 'string',
+				displayOptions: {
+					show: {
+						operation: [
+							'push',
+						],
+					},
+				},
+				typeOptions: {
+					alwaysOpenEditWindow: true,
+				},
+				default: '',
+				required: true,
+				description: 'Data to push',
+				hint: 'Value or a string of comma separated values',
+			},
+			{
+				displayName: 'Tail',
+				name: 'tail',
+				type: 'boolean',
+				displayOptions: {
+					show: {
+						operation: [
+							'push',
+							'pop',
+						],
+					},
+				},
+				default: false,
+				description: 'Whether to operate on data at the tail of the list',
+			},
+			{
+				displayName: 'Overwrite',
```

Comment:
To understand what this does, I had to look at the code. We need a better name and description for this or remove it.

Reply:
Is this referring to the `Tail` or `Overwrite`? Would `'Whether to push or pop data from the end of the list'` be more clear?

Reply:
I've removed the `Overwrite` option entirely.